---
bool success
float32 weight
string status